pipeline_dp/aggregate_params.py

```diff
@@ -113,45 +113,83 @@ def metrics_str(self) -> str:
-                                        self.max_value is None):
-                raise ValueError(
-                    "AggregateParams: min_value and max_value must be set")
-            if needs_min_max_value and (_not_a_proper_number(self.min_value) or
-                                        _not_a_proper_number(self.max_value)):
-                raise ValueError(
-                    "AggregateParams: min_value and max_value must be both "
-                    "finite numbers")
-            if needs_min_max_value and self.max_value < self.min_value:
-                raise ValueError(
-                    "AggregateParams: max_value must be equal to or greater than"
-                    " min_value")
-            if Metrics.VECTOR_SUM in self.metrics and \
-            (Metrics.SUM in self.metrics or \
-            Metrics.MEAN in self.metrics or \
-            Metrics.VARIANCE in self.metrics):
+        if (self.min_value is None) != (self.max_value is None):
+            raise ValueError(
+                "AggregateParams: min_value and max_value should be or both set or both None."
+            )
+        if (self.min_sum_per_partition is None) != (self.max_sum_per_partition
+                                                    is None):
+            raise ValueError(
+                "AggregateParams: min_sum_per_partition and max_sum_per_partition should be or both set or both None."
+            )
+
+        value_bound = self.min_value is not None
+        if value_bound:
+            self._check_range_correctness("min_value", "max_value")
+            if self.max_contributions_per_partition is None:
                 raise ValueError(
-                    "AggregateParams: vector sum can not be computed together "
-                    "with scalar metrics, like sum, mean etc")
+                    "AggregateParams: max_contributions_per_partition must be "
+                    "set when min_value/max_value are set.")
+
+        partition_bound = self.min_sum_per_partition is not None
+        if partition_bound:
+            self._check_range_correctness("min_sum_per_partition",
+                                          "max_sum_per_partition")
+
+        if value_bound and partition_bound:
+            raise ValueError(
+                "min_value and min_sum_per_partition can not be both set.")
+
+        if self.metrics:
+            if Metrics.VECTOR_SUM in self.metrics:
+                if Metrics.SUM in self.metrics or Metrics.MEAN in self.metrics or Metrics.VARIANCE in self.metrics:
+                    raise ValueError(
+                        "AggregateParams: vector sum can not be computed together"
+                        " with scalar metrics, like sum, mean etc")
+            elif partition_bound:
+                ALL_ALLOWED_METRICS = set(
+                    [Metrics.SUM, Metrics.PRIVACY_ID_COUNT, Metrics.COUNT])
+                not_allowed_metrics = set(
```

Comment:
Why is this lowercase?

Reply:
Is it OK to use lowercase for a const, then, for consistency? I also don't think it is a const conceptually, it takes different values in different branches.